src/python_inspector/resolution.py

```diff
@@ -116,24 +138,51 @@ def get_environment_marker_from_environment(environment):
     }
 
 
-def is_requirements_file_in_setup_files(setup_files: List[str]) -> bool:
+def contain_string(string: str, files: List) -> bool:
     """
-    Return True if the string ``requirements.txt`` is found in any of the ``setup_files`` location
-    strings to either a setup.py or setup.cfg file.
-    This is an indication that a requirements.txt is likely loaded in the setup.py
-    code and we use this as a hint to treat requirements.txt requirements
-    as being for the setup.py file.
+    Return True if the string is contains in any of the files.
     """
-    for setup_file in setup_files:
-        if not os.path.exists(setup_file):
+    for file in files:
+        if not os.path.exists(file):
             continue
-        with open(setup_file, encoding="utf-8") as f:
+        with open(file, encoding="utf-8") as f:
             # TODO also consider other file names
-            if "requirements.txt" in f.read():
+            if string in f.read():
                 return True
     return False
 
 
+def parse_reqs_from_setup_py_insecurely(setup_py):
+    """
+    Yield requirements from the setup.py file at ``setup_py``.
+    """
+    if not os.path.exists(setup_py):
+        return []
+    for req in iter_requirements(level="", extras=[], setup_file=setup_py):
+        yield Requirement(req)
+
+
+def parse_deps_from_setup_py_insecurely(setup_py):
+    """
+    Yield requirements from the setup.py file at ``setup_py``.
+    """
+    if not os.path.exists(setup_py):
+        return []
+    for req in iter_requirements(level="", extras=[], setup_file=setup_py):
+        parsed_req = Requirement(req)
+        yield DependentPackage(
+            purl=str(
+                PackageURL(
+                    type="pypi",
+                    name=parsed_req.name,
+                )
+            ),
+            extracted_requirement=req,
+            scope="install",
```

Comment:
Are we sure we always look exclusively for install_requires for now?

Reply:
Yes!